### PR TITLE
add sage-proteomics recipes

### PR DESCRIPTION
**Pull request by lcion-02**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/sage-proteomics/meta.yaml [hunk 1]
```diff
@@ -16,12 +16,11 @@ source:
 
 requirements:
   build:
-    - rust >=1.62
+    - {{ compiler('c') }}
   host:
-
+    - rust >=1.18.0
```
Comment (lazear) [inline recipes/sage-proteomics/meta.yaml:21]: Sage has a MSRV (minimum supported rust version) of 1.62
It will not compile on any version less than 1.62

**Comment by bgruening** [comment]
The test is now failing. So either the binary is called differently or you are not installing into the $PREFIX/bin directory. Can you check that @lcion-02 or @lazear ?

**Comment by lazear** [comment]
> The test is now failing. So either the binary is called differently or you are not installing into the $PREFIX/bin directory. Can you check that @lcion-02 or @lazear ?

@bgruening how can one test recipes locally? I have 0 conda experience...
Generally, `cargo build --release` will generate the binary at the location `$(pwd)/target/release/sage`. So that might be the issue?

File: recipes/sage-proteomics/meta.yaml [hunk 2]
```diff
@@ -18,7 +17,7 @@ requirements:
   build:
     - {{ compiler('c') }}
   host:
-    - rust >=1.18.0
+    - rust >=1.62.0
```
Comment (lazear) [inline recipes/sage-proteomics/meta.yaml:20]: This actually should be under build. No requires for rust on host

**Comment by lazear** [comment]
```yml
{% set name = "sage-proteomics" %}
{% set version = "0.7.1" %}

package:
  name: {{ name|lower  }}
  version: {{ version  }}

build:
  number: 0
  script: RUST_BACKTRACE=1 cargo install --verbose --root $PREFIX --path crates/sage-cli

source:
  url: https://github.com/lazear/sage/archive/refs/tags/v{{ version }}.tar.gz
  sha256: ba5db3d68d4bf4738b7136bc651481fc70011bd62ae4ee68a2a948faab733287

requirements:
  build:
    - {{ compiler('c') }}
    - rust >=1.62.0
  host:
  run:

test:
  commands:
    - sage --help

about:
  home: https://github.com/lazear/sage
  license: MIT
  license_file: LICENSE
  summary: |
    Proteomics searching so fast it feels like magic.
```

**Comment by lcion-02** [comment]
> Perhaps look at alevin-fry for another example of a Rust tool. They use cargo install to install the binary to the path.
> 
> https://github.com/lcion-02/bioconda-recipes/blob/sage-proteomics/recipes/alevin-fry/meta.yaml https://github.com/lcion-02/bioconda-recipes/blob/sage-proteomics/recipes/alevin-fry/build.sh

okay. I'll look at it

**Comment by lazear** [comment]
> > Perhaps look at alevin-fry for another example of a Rust tool. They use cargo install to install the binary to the path.
> > https://github.com/lcion-02/bioconda-recipes/blob/sage-proteomics/recipes/alevin-fry/meta.yaml https://github.com/lcion-02/bioconda-recipes/blob/sage-proteomics/recipes/alevin-fry/build.sh
> 
> okay. I'll look at it

Or, just look at the working meta.yml file I posted above :)

**Comment by bgruening** [comment]
Do you want to install or build the tool in the recipe?

**Pull request closed without merging**